lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -538,14 +528,22 @@ export class UserAgentApplication {
         throw ClientAuthError.createInvalidInteractionTypeError();
       }
 
-      // prompt user for interaction
-      this.navigateWindow(urlNavigate, popUpWindow);
+      let popUpWindow: Window;
+      if (interactionType === Constants.interactionTypePopup) {
+        // Generate a popup window
+        popUpWindow = this.openWindow(urlNavigate, "_blank", 1, this, resolve, reject);
+        if (!popUpWindow) {
+          // We pass reject in openWindow, we reject there during an error
+          return;
```

Comment:
My understanding is that this code is attempting to open a popup to a blank page, to check if popups are allowed (i.e. `popUpWindow` will be `null` if popups are blocked), and then short-circuits the `acquireTokenHelper` function if that happens. This ensures none that `acquireTokenAuthority.resolveEndpointsAsync` and the code in its `.then` don't run unnecessarily. I'm assuming this is done intentionally, to prevent orphaned callbacks (which are registered above).

Reply:
If the browser cannot open the link, the code hangs on line with "openWindow".

Reply:
> If the browser cannot open the link, the code hangs on line with "openWindow".

Specifically in the scenario you describe above, correct? I'm referring to scenarios such as the browser having popups blocked, for example.

Reply:
Yes. I didn't find any other scenarios when it doesn't work. I still didn't find why it hangs ;/